crates/db/types/src/mmr_helpers.rs

```diff
@@ -0,0 +1,331 @@
+//! MMR position calculation helpers
+//!
+//! This module provides bit manipulation utilities for navigating the MMR structure.
+//!
+//! # Note on Future Migration
+//!
+//! These helpers are pure MMR navigation utilities that don't depend on any
+//! Alpen-specific logic. They are candidates for upstreaming to the `strata-merkle`
+//! crate in the `strata-common` repository, where they would benefit the entire
+//! ecosystem. For now, they live here in `strata-db-types` to avoid duplication
+//! between the database layer and storage manager layer.
+//!
+//! # MMR Structure
+//!
+//! MMR (Merkle Mountain Range) uses post-order traversal numbering:
+//!
+//! ```text
+//!       7
+//!      /  \
+//!     3    6
+//!    / \  / \
+//!   1  2 4  5
+//!
+//! Position: [0, 1, 2, 3, 4, 5, 6, 7]
+//! Height:   [0, 0, 1, 0, 0, 1, 2, 0]
+//! Leaves:   [0, 1, x, 2, 3, x, x, 4]  (x = internal node)
+//! ```
+
+use crate::{DbError, DbResult};
+
+/// Convert leaf index to total MMR size (number of nodes)
+///
+/// Formula: 2 * leaves - peak_count
+/// Peak count = number of set bits in binary representation of leaf count
+///
+/// # Examples
+///
+/// ```text
+/// 1 leaf  (0b1)   -> 1 peak  -> size = 2*1 - 1 = 1
+/// 2 leaves (0b10)  -> 1 peak  -> size = 2*2 - 1 = 3
+/// 3 leaves (0b11)  -> 2 peaks -> size = 2*3 - 2 = 4
+/// 7 leaves (0b111) -> 3 peaks -> size = 2*7 - 3 = 11
+/// ```
+pub fn leaf_index_to_mmr_size(index: u64) -> u64 {
+    let leaves_count = index + 1;
+    let peak_count = leaves_count.count_ones() as u64;
+    2 * leaves_count - peak_count
+}
```

Comment:
This implementation should work only for the last (most right) index. For other indices it depends on the total number on leaves on the tree/mmr.

Let's change the name to `leave_to_mmr_size` and perhaps takes as input `leaves_count` instead?